impl/etcd/queue/etcd.go

```diff
@@ -17,29 +17,46 @@ package queue
 import (
 	"bytes"
 	"encoding/gob"
-	"errors"
 	"fmt"
 	"log"
 	"strconv"
 	"time"
 
+	"github.com/gogo/protobuf/proto"
+	"github.com/google/keytransparency/core/mutator"
 	"github.com/google/keytransparency/core/queue"
 	ctxn "github.com/google/keytransparency/core/transaction"
 	itxn "github.com/google/keytransparency/impl/transaction"
 
 	v3 "github.com/coreos/etcd/clientv3"
 	"golang.org/x/net/context"
+
+	tpb "github.com/google/keytransparency/core/proto/keytransparency_v1_types"
 )
 
-// leaseTTL is in seconds.
-var leaseTTL = int64(30)
+var (
+	// leaseTTL is in seconds.
+	leaseTTL = int64(30)
+	// saveMutation will be overwritten in tests.
+	saveMutation = func(txn ctxn.Txn, mutations mutator.Mutation, mutation []byte) error {
```

Comment:
I wasn't thinking of doing this at this stage but it seems like a good idea. However, I think I'll devote a separate PR for this. That PR will include:

* Getting rid of `Enqueue` and moving storing mutations in the database to `keyserver.go`.
* Moving `AdvanceEpoch` and its functionality to `signer.go`.

This will basically remove our dependency on the queue so we can delete it completely.